e2etest/pkg/config/validate.go

```diff
@@ -14,6 +14,13 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+func ValidateAnnotateIPAllocatedFromPool(addressPool *metallbv1beta1.IPAddressPool, annotations map[string]string) error {
+	if poolName, ok := annotations["metallb.universe.tf/ip-allocated-from-pool"]; ok && poolName == addressPool.Name {
+		return nil
+	}
+	return errors.New("annotation of service with the pool used to provide its no found")
```

Comment:
Much better :-)

Reply:
But since a miss in a map returns the zero-value of the element (in this case the string), you can even do
`if annotations["metallb.universe.tf/ip-allocated-from-pool"] == addressPool.Name`
At this point I'd argue we need a function :-)

Reply:
Do you mean it need another function here?  I don'k think so ..

`if poolName, ok := annotations["metallb.universe.tf/ip-allocated-from-pool"]; ok && poolName == addressPool.Name` it validate two things:

- Do map-key(`metallb.universe.tf/ip-allocated-from-pool`) is exist?
- If exist, checking the values of `addressPool.Name` and map-value are equal.

Reply:
Yes, but since when you check the map and the item is not there, the map will give you "", there's no need for the "ok" part

Reply:
It's not like java where you'd get a NPE, for example